config/externalname.go

```diff
@@ -47,6 +47,21 @@ var TerraformPluginFrameworkExternalNameConfigs = map[string]config.ExternalName
 	//
 	// PodIdentityAssociation can be imported using the association ID by passing spec.forProvider.clusterName field
 	"aws_eks_pod_identity_association": eksPodIdentityAssociation(),
+
+	// opensearchserverless
+	//
+	// AccessPolicy can be imported using the policy name
+	"aws_opensearchserverless_access_policy": config.NameAsIdentifier,
```

Comment:
When I check the tf registry doc, I observe that the import format is: `name and type arguments separated by a slash (/)`. This also applies to other resources with `NameAsIdentifier`. I'm not proposing to change this since the uptests were successful (and the imports were successful) but I wanted to leave this comment for the sake of understanding the approach here.

Reply:
In the no-fork architecture, we never visit the `Import` path of the terraform implementation and directly start with an `Observe` with the ID+provided parameters. In many resources, especially in plugin framework resources, you see that the ID is transformed during import. Therefore, the import ID and the "regular" ID format is frequently different. See the below example of `aws_opensearchserverless_lifecycle_policy` resource import implementation at terraform provider, which is also similar at other `NameAsIdentifier` resources:
https://github.com/upbound/terraform-provider-aws/blob/14c3fe90f23bf825d5d7301f1a7d3303c59664fc/internal/service/opensearchserverless/lifecycle_policy.go#L242-L254

As you can see from the implementation, the incoming ID with slashes is split and the first part is then assigned to both `name` and `id`. After that point, resource state continues its lifecycle as such.

Also at the Read implementation: https://github.com/upbound/terraform-provider-aws/blob/14c3fe90f23bf825d5d7301f1a7d3303c59664fc/internal/service/opensearchserverless/lifecycle_policy.go#L145

we see that the ID and type are supplied separately, where ID corresponds to name. If we were to supply the ID as the slash-separated import format, then it would have failed to `Read()` the resource, hence the `Observe` would fail and the import would never happen.

Reply:
Thank you for the detailed explanation.